cub/grid/grid_even_share.cuh

```diff
@@ -122,17 +122,18 @@ public:
      * \brief Dispatch initializer. To be called prior prior to kernel launch.
      */
     __host__ __device__ __forceinline__ void DispatchInit(
-        OffsetT num_items,          ///< Total number of input items
+        OffsetT num_items_,          ///< Total number of input items
         int     max_grid_size,      ///< Maximum grid size allowable (actual grid size may be less if not warranted by the the number of input items)
         int     tile_items)         ///< Number of data items per input tile
     {
-        this->block_offset          = num_items;    // Initialize past-the-end
-        this->block_end             = num_items;    // Initialize past-the-end
-        this->num_items             = num_items;
-        this->total_tiles           = (num_items + tile_items - 1) / tile_items;
-        this->grid_size             = CUB_MIN(total_tiles, max_grid_size);
+        this->block_offset          = num_items_;    // Initialize past-the-end
+        this->block_end             = num_items_;    // Initialize past-the-end
+        this->num_items             = num_items_;
+        this->total_tiles           = (num_items_ + tile_items - 1) / tile_items;
+        this->grid_size             = CUB_MIN(static_cast<int>(total_tiles), max_grid_size);
         OffsetT avg_tiles_per_block = total_tiles / grid_size;
-        this->big_shares            = total_tiles - (avg_tiles_per_block * grid_size);        // leftover grains go to big blocks
+        // leftover grains go to big blocks
+        this->big_shares = static_cast<int>(total_tiles - (avg_tiles_per_block * grid_size));
```

Comment:
Is there any reason `big_share` can't be an `OffsetT`?

Reply:
I took a closer look and if anything, `total_tiles` should be changed to a 32-bit int since it's used as a block dimension and those can't exceed INT32_MAX. The only usages of the variable assume it is a 32-bit int.

I'll change this so that `total_tiles` is appropriately sized, which will eliminate the need for this (and a few other) `static_cast`s.